app/questionnaire/navigation.py

```diff
@@ -98,7 +99,8 @@ def _build_single_navigation(self, section, current_group_id, first_location):
     def _build_repeating_navigation(self, repeating_rule, section, current_group_id, current_group_instance):
         group = self.schema.get_group(section['group_order'][0])
         first_location = Location(group['id'], 0, group['blocks'][0]['id'])
-        no_of_repeats = evaluate_repeat(repeating_rule, self.answer_store, self.routing_path)
+        answer_on_path = PathFinder.get_answer_ids_on_routing_path(self.schema, self.routing_path)
```

Comment:
This variable name isn't very clear, it sounds singular and possibly boolean. It would be much clearer named something like `answer_ids_on_path`